src/utils/helper.ts

```diff
@@ -0,0 +1,10 @@
+export const transformProtectionValues = (
+  data: Record<string, any>
+): Record<string, any> => {
+  const { catCave = '', catCode = '', catHandling = '', ...rest } = data
+  const item = {
+    ...rest,
+    protectionString: `${catCave}-${catCode}-${catHandling}`
```

Comment:
this needs the `protectiveMarking` field. They're in this order:
- catCode
- protectiveMarking
- catHandle
- catCave

Note: we need to allow for multiple values of catCode, catHandle and catCave, but just one for `protectiveMarking`.

Reply:
Hi @IanMayo, We have `protectiveMarking` field in items that's why we are saving only catCave, catCode and catHandling fields in this. So here do you mean we need to create a new ref table for `protectiveMarking` and then save ref id to `protectionString` field and also remove it from items/batches?

Reply:
Sorry @salony75way - I thought I'd answered this.

We are not formally storing this data in `protectionString`.  The formal place for all of this data is in the bridging tables (for cat_code, cat_handle and cat_cave), and the `protectiveMarking` for `batches` and `items`.

The `protectionString` is just a convenience for displaying this data in `BatchList` and `ItemList` (plus _maybe_ some reports).

So, we will never parse `protectionString`.  If we need the values we'll take them from the bridging table (or `protectiveMarking` field).  We will only use it to display `Protection` in `BatchList` and `ItemShow`.

Oh, but we will update the value on `save batch` or `save item` (_maybe_ only if the protection values change)